server/src/main/java/org/elasticsearch/index/fielddata/ordinals/GlobalOrdinalsBuilder.java

```diff
@@ -50,9 +52,25 @@ public static IndexOrdinalsFieldData build(
             atomicFD[i] = indexFieldData.load(indexReader.leaves().get(i));
             subs[i] = atomicFD[i].getOrdinalsValues();
         }
-        final OrdinalMap ordinalMap = OrdinalMap.build(null, subs, PackedInts.DEFAULT);
+        final TermsEnum[] termsEnums = new TermsEnum[subs.length];
+        final long[] weights = new long[subs.length];
+        final long[] counter = new long[1];
+        for (int i = 0; i < subs.length; ++i) {
+            termsEnums[i] = new FilterLeafReader.FilterTermsEnum(subs[i].termsEnum()) {
+                @Override
+                public BytesRef next() throws IOException {
+                    // check parent circuit breaker every 8192 calls
+                    if ((++counter[0] & 0x1FFF) == 0x1FFF) {
```

Comment:
I'm tempted to do the first check immediately, since `OrdinalMap` allocates a few things before it even starts iterating terms enums?

And then update the test to make sure that the circuit breaker has beet called twice?

```suggestion
                    if ((counter[0]++ & 0x1FFF) == 0) {
```

Reply:
I didn't realise this, so +1 to check immediately.